src/m365/spo/commands/file/file-list.ts

```diff
@@ -26,42 +29,67 @@ class SpoFileListCommand extends SpoCommand {
     return 'Lists all available files in the specified folder and site';
   }
 
+  
+
   public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+
     if (this.verbose) {
       logger.logToStderr(`Retrieving all files in folder ${args.options.folder} at site ${args.options.webUrl}...`);
     }
 
-    let requestUrl: string = `${args.options.webUrl}/_api/web/GetFolderByServerRelativeUrl('${encodeURIComponent(args.options.folder)}')/Files`;
 
-    if (args.options.output !== 'json') {
-      requestUrl += '?$select=UniqueId,Name,ServerRelativeUrl';
-    }
+    // Recursive function
+    const getFiles = (folderUrl: string, files: FilePropertiesCollection={value:[]}):Promise<FilePropertiesCollection> => {
```

Comment:
It would be clearer to move this to a separate function outside of `commandAction` rather than nesting it here

Reply:
Fixed in the latest commit